resource-service/src/component-tests/java/uk/gov/gchq/palisade/component/resource/repository/RedisPersistenceLayerTest.java

```diff
@@ -0,0 +1,231 @@
+                .as("Check that the list is empty as the resource has expired from the persistence")
+                .isEmpty();
+    }
+
+    @Test
+    void testPersistenceTtlForResourceFormat() throws InterruptedException {
+        // Given the resource has been added to the persistence
+
+        // When getting a resource from the persistence layer by serialisedFormat
+        var idResult = persistenceLayer.getResourcesBySerialisedFormat(resource.getSerialisedFormat())
+                .join().orElseThrow()
+                .toMat(Sink.seq(), Keep.right()).run(materialiser)
+                .toCompletableFuture()
+                .join();
+
+        // Then the returned resource should match the created resource
+        assertThat(idResult)
+                .as("Check that when getting a resource by its serialised format, the correct resource is returned")
+                .containsOnly(resource)
+                .first()
+                .usingRecursiveComparison()
+                .isEqualTo(resource);
+
+        // Then sleep to imitate a persistence timeout
+
+        TimeUnit.SECONDS.sleep(10);
+
+        // When getting a resource from the persistence layer by resourceId
+        var resourceOptional = persistenceLayer.getResourcesBySerialisedFormat(resource.getSerialisedFormat())
+                .join();
+
+        // Then no resources are returned
+        assertThat(resourceOptional)
+                .as("Check that the list is empty as the resource has expired from the persistence")
+                .isEmpty();
+    }
+
+    @Test
+    void testPersistenceTtlForInFlightResources() throws InterruptedException {
+        // When getting a resource from the persistence layer by resourceId, imitating a slow request
+        var inFlight = persistenceLayer.getResourcesById(resource.getId())
+                .join()
+                .orElseThrow();
+
+        // Then Sleep for 2 seconds to imitate a slow request
+        // The persistence evict for completeness happens here, but not for resources
+        TimeUnit.SECONDS.sleep(5);
+
+        // Now complete the retrieval of the request
+        var idResult = inFlight.runWith(Sink.seq(), materialiser)
+                .toCompletableFuture()
+                .join();
+
+        assertThat(idResult)
+                .as("Check that when getting a resource by its Id, the correct resource is returned")
+                .containsOnly(resource);
+    }
+
+
+    @Test
+    void testPersistenceTtlForExpiredInFlightResources() throws InterruptedException {
+        // When getting a resource from the persistence layer by resourceId, imitating a slow request
+        var inFlight = persistenceLayer.getResourcesById(resource.getId())
+                .join().orElseThrow();
+
+        // Then Sleep for 2 seconds to imitate a slow request
+        // The persistence evict for completeness happens here and also for resources
+        TimeUnit.SECONDS.sleep(15);
```

Comment:
While this has been _okay_ in previous tests, here we spend nearly a minute sleeping and waiting.
A better option might be to (either through reflection or extending the class in some way) set the TTL on a per-method basis, such that in each case we are waiting something like 1s for our expected ttl 'window', and the next longest expiry is suitably distant that we don't get issues.

For example, this would have a
```
// Given we have set the completeness AND resource repository ttl to 1s
// When we write resources to redis and then sleep for 1s
// Then the resources are evicted
```